packages/client/tests/functional/extensions/enabled/query.ts

```diff
@@ -862,6 +862,47 @@ testMatrix.setupTestSuite(
         `Error caused by an extension: All is lost!`,
       )
     })
+
+    test('empty args becomes an empty object', async () => {
+      const fnUser1 = jest.fn()
+      const fnEmitter = jest.fn()
+
+      prisma.$on('query', fnEmitter)
+
+      const xprisma = prisma.$extends({
+        query: {
+          user: {
+            findFirst({ args, query, operation, model }) {
+              fnUser1(args)
+              return query(args)
+            },
+          },
+        },
+      })
+
+      const args = undefined
+
+      const data = await xprisma.user.findFirst(args)
+
+      expect(data).not.toBe(null)
+      expect(fnUser1).toHaveBeenCalledWith({})
+      expect(fnUser1).toHaveBeenCalledTimes(1)
+      await waitFor(() => expect(fnEmitter).toHaveBeenCalledTimes(1))
+    })
+
+    test('passing incorrect argument errors', () => {
+      prisma.$extends({
+        query: {
+          user: {
+            async findFirst({ args, query, operation, model }) {
+              const user = await query(args)
+              // @ts-expect-error
+              return query(user)
```

Comment:
A type fix to prevent mistakes. `query` previously did not typecheck inputs (just `object`), now has full safety.